includes/Modules/Analytics_4/Web_Tag.php

```diff
@@ -43,12 +62,31 @@ public function register() {
 	protected function enqueue_gtag_script() {
 		if ( did_action( 'googlesitekit_analytics_init_tag' ) ) {
 			// If the gtag script is already registered in the Analytics module, then we need to add <MEASUREMENT_ID> configuration only.
-			$config = sprintf( 'gtag("config", "%s");', esc_js( $this->tag_id ) );
-			wp_add_inline_script( 'google_gtagjs', $config );
+			$this->add_inline_config( $this->tag_id, array() );
 		} else {
 			// Otherwise register gtag as in the Analytics module knowing that we used Measurement ID from GA4 instead of Property ID.
 			parent::enqueue_gtag_script();
 		}
 	}
 
+	/**
+	 * Gets the tag config as used in the gtag data vars.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Tag configuration.
+	 */
+	protected function get_tag_config() {
+		$config = parent::get_tag_config();
+
+		if ( ! empty( $this->custom_dimensions ) ) {
+			$config[ $this->tag_id ] = array_merge(
+				$config[ $this->tag_id ],
```

Comment:
For the regular, non-AMP version, we just want to merge the custom dimensions directly into the config (this is as per the [Design Doc](https://docs.google.com/document/d/1htqStvvyXq_eg5GsHSbdBBgm4Tu_Z6KXsIOBfjWQ5oM/edit?resourcekey=0-Px-yDiCQOTz30DBKllhU3g#heading=h.8rpbnsc62d64) but was not explicitly captured in the IB):

```suggestion
			$config = array_merge(
				$config,
```

Reply:
Oh... thanks, good catch.